### PR TITLE
Completed project - Cora Jacobson

### PR DESCRIPTION
**Pull request by CoraJacobson**
@swift-student

### CODE REVIEW

File: AirportDepartures.playground/Contents.swift [hunk 1]
```diff
@@ -16,9 +16,68 @@ import UIKit
...
+    let location: String
+}
+
+struct Flight {
+    let airline: String
+    let number: String
+    let destination: Airport
+    var terminal: String?
+    var departureTime: String?
+    var status: (FlightStatus, Timing?)
```
Comment (swift-student) [inline AirportDepartures.playground/Contents.swift:46]: Very nice use of a tuple here! The only thing I would suggest is possibly naming the elements of the tuple so that it is more clear when you use them later, such as line 63. I can't think of what you would name the first element though, except maybe just flightStatus. So you would have status.flightStatus and status.timing. It's not too hard to follow with just two un-named elements though.

File: AirportDepartures.playground/Contents.swift [hunk 2]
```diff
@@ -16,9 +16,68 @@ import UIKit
...
+                case .landed:
+                    print("\(flight.airline) Flight \(flight.number) has landed at Terminal: \(flight.terminal ?? "TBD")")
+                case .boarding:
+                    print("\(flight.airline) Flight \(flight.number) is boarding, please head to Terminal: \(flight.terminal ?? "TBD") immediately. The doors are closing soon.")
+                case .departed:
+                    print("\(flight.airline) Flight \(flight.number) has departed from Terminal: \(flight.terminal ?? "unknown")")
+                }
+            }
+        }
+    }
```
Comment (swift-student) [inline AirportDepartures.playground/Contents.swift:79]: Nicely written alert function. The switch statement makes things very easy to read.

File: AirportDepartures.playground/Contents.swift [hunk 3]
```diff
@@ -29,9 +88,40 @@ import UIKit
...
+    let userCalendar = Calendar.current
+    let time = userCalendar.date(from: timeComponents)
+    if let unwrappedTime = time {
+        let formatter = DateFormatter()
+        formatter.timeStyle = .short
+        return formatter.string(from: unwrappedTime)
+    } else {
+        return "TBD"
+    }
+}
```
Comment (swift-student) [inline AirportDepartures.playground/Contents.swift:112]: Nice helper function! You are thinking like a developer here.

File: AirportDepartures.playground/Contents.swift [hunk 4]
```diff
@@ -58,9 +153,29 @@ import UIKit
...
+            if let terminal = flight.terminal {
+                print("Destination: \(flight.destination.name) - \(flight.destination.abbreviation)  Airline: \(flight.airline)  Flight #: \(flight.number)  Departure Time:  Terminal: \(terminal)  Status: \(flight.status.0.rawValue)\(flight.status.1?.rawValue ?? "")")
+            }
+        } else if let time = flight.departureTime {
+            if let terminal = flight.terminal {
+                print("Destination: \(flight.destination.name) - \(flight.destination.abbreviation)  Airline: \(flight.airline)  Flight #: \(flight.number)  Departure Time: \(time)  Terminal: \(terminal)  Status: \(flight.status.0.rawValue)\(flight.status.1?.rawValue ?? "")")
+            }
+        }
+    }
+}
```
Comment (swift-student) [inline AirportDepartures.playground/Contents.swift]: I would challenge you to see if you can do this with only one print statement. One way to approach it would be to set up a variable to a default value, and if you can unwrap your optional, set that variable to the unwrapped value.

``` swift
var terminal = ""
if let flightTerminal = flight.terminal {
    terminal = flightTerminal
}

// terminal is now equal to either an empty string "" or the unwrapped flight terminal
```

File: AirportDepartures.playground/Contents.swift [hunk 5]
```diff
@@ -96,6 +209,17 @@ import UIKit
...
+func calculateAirfare(checkedBags: Int, distance: Int, travelers: Int) -> String {
+    let airfare = ((Double(checkedBags) * 25) + (Double(distance) * 0.10)) * Double(travelers)
+    let formatter = NumberFormatter()
+    formatter.numberStyle = .currency
+    if let formattedAirfare = formatter.string(for: airfare) {
+        return formattedAirfare
+    } else {
+        return "invalid input"
+    }
+}
```
Comment (swift-student) [inline AirportDepartures.playground/Contents.swift:215]: Very nice work! Great use of the formatter, and defaulting to "invalid input"

**Review by swift-student — COMMENTED** [review]
Cora, you demonstrated that you are understanding the concepts extremely well. Overall, another solid submission. You should have no problem with the sprint challenge this weekend!

**Comment by CoraJacobson** [comment]
@swift-student 
Revised Step 4
